src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -229,6 +232,52 @@ export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
     }
   }
 
+  const preparePoolKey = (
+    usecase: LedgerTypes.TransactionSigningMode,
+    poolKeyHash: Buffer,
+    poolKeyPath?: BIP32Path,
+  ): LedgerTypes.PoolKey => {
+    if (usecase === LedgerTypes.TransactionSigningMode.POOL_REGISTRATION_AS_OPERATOR) {
+      return {
+        type: LedgerTypes.PoolKeyType.DEVICE_OWNED,
+        params: {
+          path: poolKeyPath as BIP32Path,
+        },
+      }
+    }
+    if (usecase === LedgerTypes.TransactionSigningMode.POOL_REGISTRATION_AS_OWNER) {
+      return {
+        type: LedgerTypes.PoolKeyType.THIRD_PARTY,
+        params: {
+          keyHashHex: poolKeyHash.toString('hex'),
+        },
+      }
+    }
+    throw Error(Errors.InvalidTransactionType)
+  }
+
+  const prepareRewardAccount = (
+    signingFiles: HwSigningData[],
+    rewardAccount: Buffer,
+    network: Network,
+  ): LedgerTypes.PoolRewardAccount => {
+    const addressParams = getAddressParameters(signingFiles, rewardAccount, network)
+    if (addressParams && addressParams.address === rewardAccount) {
```

Comment:
I don't understand the logic of this. If I turn a reward address into adress params, and the params do not correspond to reward address, it means something went very wrong, doesn't it? So I would expect an exception being thrown or a failed assertion, not just happily returning some value.

Reply:
I agree, the whole `getAddressParameters` method logic is weird. The address match/mismatch is checked everywhere outside of this method. I think it makes more sense to just return null when mismatch occurs, so I refactored it in this way.

When getting address params, they are constructed with provided hwsfiles, so in case of null we just assume it's not a device owned address.